test/hotspot/jtreg/serviceability/jvmti/events/Exception/exception01/libexception01.cpp

```diff
@@ -0,0 +1,317 @@
+  if (err != JVMTI_ERROR_NONE) {
+    LOG("(GetClassSignature#t) unexpected error: %s (%d)\n",
+           TranslateError(err), err);
+    result = STATUS_FAILED;
+    return;
+  }
+  err = jvmti->GetMethodName(method, &ex.t_name, &ex.t_sig, &generic);
+  if (err != JVMTI_ERROR_NONE) {
+    LOG("(GetMethodName#t) unexpected error: %s (%d)\n",
+           TranslateError(err), err);
+    result = STATUS_FAILED;
+    return;
+  }
+  ex.t_loc = location;
+  err = jvmti->GetMethodDeclaringClass(catch_method, &cls);
+  if (err != JVMTI_ERROR_NONE) {
+    LOG("(GetMethodDeclaringClass#c) unexpected error: %s (%d)\n",
+           TranslateError(err), err);
+    result = STATUS_FAILED;
+    return;
+  }
+  err = jvmti->GetClassSignature(cls, &ex.c_cls, &generic);
+  if (err != JVMTI_ERROR_NONE) {
+    LOG("(GetClassSignature#c) unexpected error: %s (%d)\n",
+           TranslateError(err), err);
+    result = STATUS_FAILED;
+    return;
+  }
+  err = jvmti->GetMethodName(catch_method,
+                                 &ex.c_name, &ex.c_sig, &generic);
+  if (err != JVMTI_ERROR_NONE) {
+    LOG("(GetMethodName#c) unexpected error: %s (%d)\n",
+           TranslateError(err), err);
+    result = STATUS_FAILED;
+    return;
+  }
+  ex.c_loc = catch_location;
+  LOG(">>> %s\n", ex.name);
+  LOG(">>>   thrown at %s.%s%s:0x%x%08x\n",
+         ex.t_cls, ex.t_name, ex.t_sig,
+         (jint)(ex.t_loc >> 32), (jint)ex.t_loc);
+  LOG(">>>    catch at %s.%s%s:0x%x%08x\n",
+         ex.c_cls, ex.c_name, ex.c_sig,
+         (jint)(ex.c_loc >> 32), (jint)ex.c_loc);
+  LOG(">>> ... done\n");
+
+
+  for (i = 0; i < sizeof(exs)/sizeof(exceptionInfo); i++) {
+    if (ex.name != NULL && strcmp(ex.name, exs[i].name) == 0
+        && ex.t_cls != NULL && strcmp(ex.t_cls, exs[i].t_cls) == 0
+        && ex.t_name != NULL && strcmp(ex.t_name, exs[i].t_name) == 0
+        && ex.t_sig != NULL && strcmp(ex.t_sig, exs[i].t_sig) == 0
+        && ex.c_cls != NULL && strcmp(ex.c_cls, exs[i].c_cls) == 0
+        && ex.c_name != NULL && strcmp(ex.c_name, exs[i].c_name) == 0
+        && ex.c_sig != NULL && strcmp(ex.c_sig, exs[i].c_sig) == 0
+        && ex.t_loc == exs[i].t_loc && ex.c_loc == exs[i].c_loc) {
+      jboolean isVirtual = jni->IsVirtualThread(thr);
+      if (isVirtualExpected != isVirtual) {
+        LOG("The thread IsVirtualThread %d differs from expected %d.\n", isVirtual, isVirtualExpected);
+        result = STATUS_FAILED;
+      } else {
+        eventsCount++;
+      }
+      break;
+    }
+  }
+  if (i == sizeof(exs)/sizeof(exceptionInfo)) {
+    LOG("Unexpected exception event:\n");
+    LOG("  %s\n", ex.name);
+    LOG("    thrown at %s.%s%s:0x%x%08x\n",
+           ex.t_cls, ex.t_name, ex.t_sig,
+           (jint)(ex.t_loc >> 32), (jint)ex.t_loc);
+    LOG("     catch at %s.%s%s:0x%x%08x\n",
+           ex.c_cls, ex.c_name, ex.c_sig,
+           (jint)(ex.c_loc >> 32), (jint)ex.c_loc);
+    result = STATUS_FAILED;
```

Comment:
Unaligned lines in the range: 172-177.
There are some non-uinified unneeded spaces at lines 172,175.